docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+```
+./gradlew startWorker
+```
+
+When the Worker starts it begins polling the Task Queue. The first Task it finds is the Workflow Task, which it executes. Executing the Workflow causes the Activity tasks to be sent to the Task Queue as well. The Worker then pulls each of the Activity Tasks in their respective order from the Task Queue and executes the corresponding Activities.
+
+Congratulations, you have run a Temporal Workflow application!
+
+## Failure simulation
+
+![Warning](/img/docs/warning.png)
+
+By running the application as is, at minimum you got a taste of one of Temporal's amazing value propositions: visibility into the Workflow and the status of the Workers executing the code. Let's explore another key value proposition, maintaining the state of a Workflow, by simulating some failures. Make sure your Worker process is stopped before proceeding.
+
+### Server crash
+
+In most modern application environments, to accommodate a host failure, leader election processes are applied to multiple application instances which are backed by databases that maintain the application state and so on. With Temporal, even if the Temporal server is down, the state of the Workflow is preserved. You can easily test this by following these steps:
+
+1. Start the Workflow again.
+2. Verify the Workflow is running in the UI.
+3. Shut down the Temporal server by either using 'Ctrl c' or via the Docker dashboard.
+4. After it has stopped, restart the Temporal server and visit the UI.
+
+The Workflow is still there!
+
+### Activity error
+
+Next let's simulate a bug in the Activity method. Inside your project find the AccountActivityImpl.java file and uncomment the line that throws an exception in the deposit method:
+
+<!--SNIPSTART java-project-template-activity-implementation-->
+<!--SNIPEND-->
+
+Save it and run the Workflow and the Worker. The Worker executed the Workflow, and completes the `withdraw()` Activity. But then the exception is thrown when it attempts the `deposit()` Activity. Notice how the Worker keeps retrying the Activity? Visit the UI and click on the RunId of the Workflow. You will see the pending Activity listed there with details such as its state, the number of times it has been attempted, and the next scheduled attempt.
+
+![Activity UI error details](/img/docs/web-ui-activity-error-info.png)
+
+[Timeout configurations](/docs/activities/#timeouts) and [retry policies](/docs/activities/#retries) for Activities are specified in the Workflow code as Activity options. This is another key value proposition that Temporal offers, as other modern applications implement timeout and retry logic within the business process itself. In our Workflow code you can see that we have specified a setStartToCloseTimeout for our Activities, but have not specified a retry policy. That is because Temporal automatically uses a default retry policy if one isn't specified!
```

Comment:
suggestion (no big deal if not accepted) - add a code snippet here to show how to add a retry policy, as this is a VERY core part of how we are pitching temporal

Reply:
Ya, good point. I updated the Workflow code to include a retry policy for activities and updated this sentence as well.